deflate_p.h

```diff
@@ -12,7 +12,37 @@
 /* Forward declare common non-inlined functions declared in deflate.c */
 
 #ifdef ZLIB_DEBUG
-void check_match(deflate_state *s, Pos start, Pos match, int length);
+/* ===========================================================================
+ * Check that the match at match_start is indeed a match.
+ */
+static void check_match(deflate_state *s, Pos start, Pos match, int length) {
```

Comment:
Non-inline functions should not be defined in header files. 

If defined in header file, will make private copy of the function in each translation unit where included (and used).

Reply:
We can make it inline. The function is only for debug builds anyways.

Reply:
Inlining should decrease binary size by few bytes and make some compilers (and indirectly linkers) happy that don't by default inline small functions (read: Visual C++).

Reply:
Done.